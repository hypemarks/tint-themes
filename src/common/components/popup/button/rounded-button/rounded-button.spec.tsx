import { render } from '@testing-library/react';
import React from 'react';
import RoundedButton from './rounded-button';

describe('Test <RoundedButton /> Component', () => {
  it('should render snapshot', () => {
    const wrapper = render(<RoundedButton onClick={jest.fn} radius={22} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should render RoundedButton component with correct styles', () => {
    const radius = 22;
    const styles = { height: `${radius}px`, width: `${radius}px` };
    const wrapper = render(
      <RoundedButton onClick={jest.fn} radius={radius}>
        rounded-button
      </RoundedButton>
    );
    const button = wrapper.queryByText('rounded-button');
    expect(button).toHaveStyle(styles);
  });
});
