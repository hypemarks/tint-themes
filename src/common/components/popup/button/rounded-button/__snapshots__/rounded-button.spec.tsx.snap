// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <RoundedButton /> Component should render snapshot 1`] = `
<DocumentFragment>
  <button
    class="base-button rounded-button undefined "
    data-testid="base-button"
    style="height: 22px; width: 22px; border-radius: 22px;"
    type="button"
  />
</DocumentFragment>
`;
